src/inc/clrconfigvalues.h

```diff
@@ -651,6 +651,7 @@ RETAIL_CONFIG_DWORD_INFO(EXTERNAL_TieredCompilation, W("TieredCompilation"), 1,
 RETAIL_CONFIG_DWORD_INFO(UNSUPPORTED_TieredCompilation_Tier1CallCountThreshold, W("TieredCompilation_Tier1CallCountThreshold"), 30, "Number of times a method must be called after which it is promoted to tier 1.")
 RETAIL_CONFIG_DWORD_INFO(UNSUPPORTED_TieredCompilation_Tier1CallCountingDelayMs, W("TieredCompilation_Tier1CallCountingDelayMs"), 100, "A perpetual delay in milliseconds that is applied to tier 1 call counting and jitting, while there is tier 0 activity.")
 RETAIL_CONFIG_DWORD_INFO(UNSUPPORTED_TieredCompilation_Tier1DelaySingleProcMultiplier, W("TieredCompilation_Tier1DelaySingleProcMultiplier"), 10, "Multiplier for TieredCompilation_Tier1CallCountingDelayMs that is applied on a single-processor machine or when the process is affinitized to a single processor.")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_TieredCompilation_DisableTier0Jit, W("TieredCompilation_DisableTier0Jit"), 0, "For methods that don't have pregenerated code, disable jitting them at tier 0 and start with a higher tier instead. For methods that have pregenerated code, tiering occurs normally.")
```

Comment:
Could we leave this at unsupported for now, and without runtimeconfig support? I get leary about handing people too many knobs because once we give them out its hard to ever remove them. I'm hoping for example that automatically upgrading loop methods to tier1 eliminates most of the motivation anyone would have had to use this knob.

Reply:
- @mjsabby wanted to use this and wanted a supported option (https://github.com/dotnet/coreclr/issues/21856)
- I'd like to use this mode as the default (for now) for microbenchmarks in the performance repo
- For scenarios that don't mind giving up some startup time, it may be a reasonable option to use despite automatically upgrading loop methods, for instance to avoid spending two JIT cycles on methods

Reply:
I agree that eventually this option hopefully will have little value and people won't be compelled to use it

Reply:
And once it reaches a stage where it adds zero value we can just ignore the option

Reply:
It is unlikely that it would have zero value. Even if it were possible to pregenerate code for all methods in an image, it still would not cover dynamic methods if we were to enable tiering for those. That's ok though, if there will always be a reason for having the switch. I'm fine it starting it off as unsupported unless @mjsabby needs a supported option.

Tests in the perf repo are runnable with a script but they are also meant to be easily runnable by anyone with just `dotnet run` and arguments. Environment variables are often inconvenient for configuring things.

Reply:
@mjsabby, regarding your request in https://github.com/dotnet/coreclr/issues/21856:
> There already exists such a setting `TieredCompilation_Test_OptimizeTier0`, but it is marked as unsupported. So the change here may just be exposing that as a supported setting.

Do you need a supported option for the scenario you have in mind, or is it just for testing purposes?

Reply:
As I recall Test_OptimizeTier0 jits tier0 with the tier1 flag. It does not suppress call-counting or prevent the method from being 'upgraded' to tier1 and jitted a second time. Decent chance it isn't the behavior you wanted.

Reply:
yeah, it will work in retail builds just fine. Behavior-wise UNSUPPORTED and EXTERNAL are identical, the only difference is just messaging, and likely documentation. 'Upgrading' it means we decide it deserves to be supported, we change the term in the source, and we potentially document something about it on MSDN.

Reply:
It's really up to you and @kouvel then. I can't imagine large workloads that have frequent process cycles to mot want this option if startup perf is not a consideration and given that S.P.C is R2R now so having the hot methods rejitted is beneficial.

I would say that having this as a first class option would be preferred because of those reasons.

I'm almost certain of this answer but with this option the code generated does not contain the call counter code correct?

Reply:
Methods with pregenerated code would be counted in order to transition them to tier 1, methods without pregenerated code would not be counted. I'll go ahead and mark it as unsupported for now, we can always choose to make it a supported option later.